cmd/slurmCheck/slurmCheck.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	//"io"
+	"log"
+	"strings"
+	//"time"
+)
+
+type slurmCheckArray struct {
+	begin      string // whatever is in front of the squiggles
+	outToCheck string // this is the file to be checked
+	checkType  string // this is the code for how we check the output files
+	end        string // whatever is after the squiggles
+}
+
+//parseTheInput parses each line of the input file
+func parseTheInput(slurmArrayFancy string) []slurmCheckArray {
+	inFancy := fileio.EasyOpen(slurmArrayFancy)
+	var slurmArray []slurmCheckArray
+	var doneReading bool
+	var err error
+	var line string
+
+	for line, doneReading = fileio.EasyNextLine(inFancy); !doneReading; line, doneReading = fileio.EasyNextLine(inFancy) {
+
+		// fatal if there is an empty line in the file.
+		if len(line) == 0 {
+			log.Fatal("empty line in file, please remove and rerun.")
+		}
+		currentLine := processFancySlurmLine(line)
+		slurmArray = append(slurmArray, currentLine)
+	}
+
+	err = inFancy.Close()
+	exception.PanicOnErr(err)
+	return slurmArray
+}
+
+// processFancySlurmLine is a helper function to parseTheInput
+func processFancySlurmLine(lineToProcess string) slurmCheckArray {
+	var slurmLineParsed slurmCheckArray
+	var parsedFirstSquiggle, parsedSecondSquiggle []string
+
+	//parse on the first squiggle into a slice of strings
+	parsedFirstSquiggle = strings.Split(lineToProcess, "{")
+	slurmLineParsed.begin = parsedFirstSquiggle[0]
+
+	//parse on the second squiggle into a slice of strings
+	parsedSecondSquiggle = strings.Split(parsedFirstSquiggle[1], "}")
+
+	//separate the string into "words" aka fields. White space defined the boundary of each field
+	fields := strings.Fields(parsedSecondSquiggle[0])
+
+	//third field will be the type of check to perform
+	slurmLineParsed.checkType = fields[2]
+
+	//fourth field is the output file we will be checking
+	slurmLineParsed.outToCheck = fields[3]
+
+	slurmLineParsed.end = parsedSecondSquiggle[1]
```

Comment:
this line will panic if the output file is at the end of the cmd line. you can check the len(parsedSecondSquiggle) as the conditional to avoid this.